Add the capacity to use mTLS for postgresql authentication

qdelmas-flowdesk opened the pull request:
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Support mTLS authentication in the `MaterializedPostgreSQL` database engine.

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

qdelmas-flowdesk committed:
Add the capacity to use mTLS for postgresql authentication

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/ClickHouse/ClickHouse?pullRequest=79681) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/ClickHouse/ClickHouse?pullRequest=79681) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ClickHouse/ClickHouse?pullRequest=79681) it.</sub>

GrigoryPervakov commented:
Please add tests
Check out tests/integration/test_postgresql_database_engine

It's also good to support passing these settings via named collection, see StoragePostgreSQL::processNamedCollectionResult

qdelmas-flowdesk committed:
Add tests for mTLS

qdelmas-flowdesk commented:
Added a test

qdelmas-flowdesk committed:
Fix CI

qdelmas-flowdesk committed:
Wait for postgres' start

GrigoryPervakov commented on tests/integration/test_postgresql_database_engine/test.py:
```diff
@@ -46,12 +46,40 @@ def started_cluster():
+    cursor = conn.cursor()
+
+    node1.query(
+        f"CREATE DATABASE postgres_database ENGINE = PostgreSQL('postgres1-mtls:5432', 'postgres_database', 'postgres', '{pg_pass}')"
```
It should be `ENGINE = MaterializedPostgreSQL` with supplied mTLS settings

alexey-milovidov committed:
Delete aspell-dict.txt (removed in master)

`aspell-dict.txt` was deleted in master along with the `aspell` and
`codespell` style checks. Removing the file here so the PR can merge
cleanly without a modify/delete conflict.
